schedulerTools.sh

```diff
@@ -46,9 +46,17 @@ function launch_job {
   local err=`grep '^\#SBATCH --error=' ${script} | sed 's/.*error=//g'`
 
   # submit SLURM job
-  local res=`sbatch ${script}`
+  local res=$(sbatch ${script} 2>&1)
   if [ $? -ne 0 ] ; then
-      exitError 7205 ${LINENO} "problem submitting SLURM batch job"
+      exitError 7204 ${LINENO} "problem submitting SLURM batch job"
+  fi
+  if [[ $res == *"QOSMaxSubmitJobPerUserLimit"* ]]; then
+      echo "Partition chosen has to many jobs from us already, trying again with the normal partition."
+      sed -i 's|--partition=.*|--partition=normal|g' ${script}
+      res=`sbatch ${script}`
+      if [ $? -ne 0 ] ; then
+	  exitError 7205 ${LINENO} "problem re-submitting SLURM batch job in normal queue"
+      fi
   fi
```

Comment:
This should probably be...
```
res=$(sbatch ${script} 2>&1)
status=$?
if [[ $res == *"QOSMaxSubmitJobPerUserLimit"* ]]; then
    echo "Partition chosen has to many jobs from us already, trying again with the normal partition."
    sed -i 's|--partition=.*|--partition=normal|g' ${script}
    res=`sbatch ${script}`
    if [ $? -ne 0 ] ; then
        exitError 7205 ${LINENO} "problem re-submitting SLURM batch job in normal queue"
    fi
else
    if [ $status -ne 0 ] ; then
        exitError 7204 ${LINENO} "problem submitting SLURM batch job (${status}, ${res})"
    fi
fi
```
...since otherwise the `if [ $? -ne 0 ]` will fail in case the `sbatch` command hits an error.

Reply:
huh, I tested it on piz daint and it did work, the $? -ne 0 did not catch the failed submission

Reply:
That's why I moved the checking to after that

Reply:
I've taken your suggestion, though still wonder what the difference was in test case, because I did find it odd the failed submission actually did not make the status a failed status -- it's only the check for a batch job id that realizes it didn't work

Reply:
Weird. I copy & pasted your code into a standalone bash script that continuously submitted jobs (until failure). When hitting the maximum allowable limit I got the `problem submitting SLURM batch job (${status}, ${res})` error message.